src/algebra/group/defs.lean

```diff
@@ -305,18 +291,20 @@ class add_cancel_monoid (M : Type u)
   extends add_left_cancel_monoid M, add_right_cancel_monoid M
 
 /-- A monoid in which multiplication is cancellative. -/
-@[protect_proj, ancestor left_cancel_monoid right_cancel_monoid, to_additive add_cancel_monoid]
+@[protect_proj, ancestor left_cancel_monoid right_cancel_monoid]
 class cancel_monoid (M : Type u) extends left_cancel_monoid M, right_cancel_monoid M
+attribute [to_additive add_cancel_monoid] cancel_monoid
 
 /-- Commutative version of add_cancel_monoid. -/
-@[protect_proj, ancestor add_left_cancel_comm_monoid add_right_cancel_comm_monoid]
-class add_cancel_comm_monoid (M : Type u)
-  extends add_left_cancel_comm_monoid M, add_right_cancel_comm_monoid M
+@[protect_proj, ancestor add_right_cancel_monoid add_left_cancel_monoid add_comm_monoid]
+class add_cancel_comm_monoid (M : Type u) extends
+  add_right_cancel_monoid M, add_left_cancel_monoid M, add_comm_monoid M
```

Comment:
We should probably make these inherit from `add_cancel_monoid`, since `add_cancel_monoid` is the union of `add_left_cancel_monoid` and `add_right_cancel_monoid`.
```suggestion
@[protect_proj, ancestor add_cancel_monoid add_comm_monoid]
class add_cancel_comm_monoid (M : Type u) extends
  add_right_cancel_monoid M, add_left_cancel_monoid M, add_comm_monoid M
```